### ISSUE TITLE
bug(mat-selection-list): A11y - VoiceOver with Chrome doesn't read other list items when one is selected

### ISSUE THREAD

**Issue by jacksonopp** [angular/components#23836]
#### Reproduction

Link to stackblitz: https://stackblitz.com/edit/angular-j7akl8?file=src/app/list-selection-example.html 

Steps to reproduce:
1. Open stackblitz
2. Enable voiceover (System Preferences -> Accessibility -> VoiceOver -> check 'Enable VoiceOver')
3. Select the **Boots** option with either mouse or keyboard
4. Use up or down arrow to navigate through list to an unselected item
 
 
#### Expected Behavior

VoiceOver should read "[item] not selected" with [item] replaced with the currently selected item

#### Actual Behavior

VoiceOver reads "Boots selected" for any item selected.

#### Notes

I have not tested this behavior with other operating systems or screen readers.

#### Environment

- Angular: 12.2.11
- CDK/Material: 12.2.11
- Browser(s): Chrome 94.0.4606.81
- Operating System (e.g. Windows, macOS, Ubuntu): macOS 10.15.7
